index.d.ts

```diff
@@ -61,6 +61,22 @@ export interface ActionCmd<A extends Action> {
   simulate(): A;
 }
 
+export interface SetTimeoutCmd<A extends Action = never> {
+  readonly type: 'SET_TIMEOUT';
+  readonly nestedCmd: CmdType;
+  readonly delayMs: number;
+  readonly scheduledActionCreator?: (timerId: number) => A;
+  simulate(timerId: number, nestedSimulation?: CmdSimulation | MultiCmdSimulation): A[] | A | null
+}
+
+export interface SetIntervalCmd<A extends Action = never> {
```

Comment:
Do we support passing arguments, similar to `RunCmd`, for the delayed commands? Both `setTimeout` and `setInterval` support an arbitrary set of args, which get sent to the callback function, so I think it would make sense to support `args` in these commands as well.

example
```
setTimeout((message, count) => console.log(message, count), 500, 'hello world', 7);
```

Reply:
Or.. Perhaps that doesn't make sense, since this is delaying other commands. We could just delay a `RunCmd` to get args. It might make sense to support `args` to pass to the `scheduleActionCreator` however. What do you guys think?

Reply:
Yea I think the callback args aren't really necessary. Since this is taking other Cmds, we can just pass those args directly to the cmd if needed. And for passing args to the action creator, we can just use arrow functions like we've done with Cmd.run.


```ts
const runCmd = Cmd.run(
  (message, count) => console.log(message, count),
  {args: ['hello world', 500], successActionCreator: whatever}
)
return loop(newState, Cmd.setTimeout(runCmd, 1234, {scheduledActionCreator: id => actionCreator(id, arg, arg2)}))
```